packages/contracts-bedrock/src/safe/DeputyPauseModule.sol

```diff
@@ -0,0 +1,211 @@
+        try _deputyGuardianModule.deputyGuardian() returns (address) {
+            // Actual response doesn't matter here.
+        } catch {
+            revert DeputyPauseModule_InvalidDeputyGuardianModule();
+        }
+
+        // Sanity check that the SuperchainConfig is actually a SuperchainConfig.
+        // Prevents some accidental misconfigurations.
+        try _superchainConfig.guardian() returns (address) {
+            // Actual response doesn't matter here.
+        } catch {
+            revert DeputyPauseModule_InvalidSuperchainConfig();
+        }
+
+        // Initialize the immutable variables.
+        FOUNDATION_SAFE = _foundationSafe;
+        DEPUTY_GUARDIAN_MODULE = _deputyGuardianModule;
+        SUPERCHAIN_CONFIG = _superchainConfig;
+        DEPUTY = _deputy;
+    }
+
+    /// @notice Getter function for the Foundation Safe address.
+    /// @return foundationSafe_ Foundation Safe address.
+    function foundationSafe() public view returns (Safe foundationSafe_) {
+        foundationSafe_ = FOUNDATION_SAFE;
+    }
+
+    /// @notice Getter function for the DeputyGuardianModule address.
+    /// @return deputyGuardianModule_ DeputyGuardianModule address.
+    function deputyGuardianModule() public view returns (IDeputyGuardianModule deputyGuardianModule_) {
+        deputyGuardianModule_ = DEPUTY_GUARDIAN_MODULE;
+    }
+
+    /// @notice Getter function for the SuperchainConfig address.
+    /// @return superchainConfig_ SuperchainConfig address.
+    function superchainConfig() public view returns (ISuperchainConfig superchainConfig_) {
+        superchainConfig_ = SUPERCHAIN_CONFIG;
+    }
+
+    /// @notice Getter function for the deputy address.
+    /// @return deputy_ Deputy address.
+    function deputy() public view returns (address deputy_) {
+        deputy_ = DEPUTY;
+    }
+
+    /// @notice Getter function for the Pause message typehash.
+    /// @return pauseMessageTypehash_ Pause message typehash.
+    function pauseMessageTypehash() public pure returns (bytes32 pauseMessageTypehash_) {
+        pauseMessageTypehash_ = PAUSE_MESSAGE_TYPEHASH;
+    }
+
+    /// @notice Getter function for the DeputyAuth message typehash.
+    /// @return deputyAuthMessageTypehash_ DeputyAuth message typehash.
+    function deputyAuthMessageTypehash() public pure returns (bytes32 deputyAuthMessageTypehash_) {
+        deputyAuthMessageTypehash_ = DEPUTY_AUTH_MESSAGE_TYPEHASH;
+    }
+
+    /// @notice Calls the Foundation Safe's `execTransactionFromModuleReturnData()` function with
+    ///         the arguments necessary to call `pause()` on the Security Council Safe, which will
+    ///         then cause the Security Council Safe to trigger SuperchainConfig pause.
+    ///         Front-running this function is completely safe, it'll pause either way.
+    /// @param _nonce Signature nonce.
+    /// @param _signature ECDSA signature.
+    function pause(bytes32 _nonce, bytes memory _signature) external {
```

Comment:
Is there is a reason that the nonce is a bytes32 instead of the uintX?

Reply:
Hmmm I guess it works either way

Reply:
a `uintX` would suggest to me that there is a global nonce which is incremented, since we don't have that I think `bytes32` makes sense.